IMPLEMENTATION_DESIGN.md

```diff
@@ -51,21 +51,28 @@ and config should be taken into account, applies the effective config.
 The effective config contains what is called the "config sections", which are the YAML keys that can exist under projects
 in the config YAML, for example: `deploy_keys`, `secret_variables`, `group_secret_variables` and so on.
 
-Those config section are processed by the code in the methods with the `process_*` naming pattern.
+Those config sections are processed by the code in the classes inheriting from the `AbstractProcessor` class. This class
```

Comment:
Since we have an agreement, I've updated the implementation design to reflect the new structure of the code.

Reply:
👍